### PR TITLE
Account.Code bug found and fixed. Tests are updated

### PR DESCRIPTION
**Pull request by evgensheff**

### CODE REVIEW

File: db/substate_db_test.go [hunk 1]
```diff
@@ -121,6 +120,8 @@ func createDbAndPutSubstate(dbPath string) (*substateDB, error) {
 	h2 := types.Hash{}
 	h2.SetBytes(nil)
 
+	tt := &testSubstate
+	_ = tt
```
Comment (petr-hanzl) [inline db/substate_db_test.go:124]: Why is this necessary?

**Review by petr-hanzl — APPROVED** [review]

**Review by wsodsong — APPROVED** [review]
